test/analytics/__fixtures__/plugins/analytics_plugin_a/public/plugin.ts

```diff
@@ -52,10 +52,20 @@ export class AnalyticsPluginA implements Plugin {
       plugin: 'analyticsPluginA',
       step: 'setup',
     });
-
+    let found = false;
     window.__analyticsPluginA__ = {
-      getLastActions: async (takeNumberOfActions: number) =>
-        firstValueFrom(this.actions$.pipe(take(takeNumberOfActions), toArray())),
+      getLastActions: async () =>
+        firstValueFrom(
+          this.actions$.pipe(
+            takeWhile(() => !found),
+            tap(({ action, meta }) => {
+              found =
+                action === 'reportEvents' &&
+                meta.find((event: Event) => event.event_type === 'test-plugin-lifecycle');
+            }),
```

Comment:
I'm wondering now... would this require the next event to occur to validate this? Should we change this to an observable to avoid that additional wait in the tests?

```suggestion
            takeUntil(found$), // with `const found$ = new Subject<boolean>();`
            tap(({ action, meta }) => {
              if (action === 'reportEvents' &&
                meta.find((event: Event) => event.event_type === 'test-plugin-lifecycle')) {
                found$.next(true);
              } 
            }),
```

Reply:
Um, that doesn't seem to work. Maybe this change can be an improvement/fix _if_ we need it?